src/main/app/pay/payment.ts

```diff
@@ -10,16 +10,16 @@ class Link {
 
 class Links {
   @Type(() => Link)
-  readonly self: Link
+  readonly self?: Link
   @Type(() => Link)
   readonly next_url: Link
 }
 
 export class Payment {
   reference: string
-  amount: number
+  amount?: number
```

Comment:
Is that because of the mock implementation or real API does not return an amount in some cases?

Reply:
I didn't change any types because of mock implementation, I think it's not present in the response which extends this class